docs/design.md

```diff
@@ -0,0 +1,24 @@
+## High Level Deployment
+CStor Pool Auto controller will consist of following deployments:
+
+### Core Deployment
```

Comment:
Similarly, we can add some info to show what the core deployment is actually trying to do. Maybe some use case diagram that links trigger initiated by a user/or any components.